core/input.go

```diff
@@ -50,6 +50,7 @@ type InputResponse struct {
 }
 
 type InputUser struct {
+	userId                 string			`json:"roles,omitempty"`
```

Comment:
This must be exported in order to Work, and the json tag must be changed as well; also you must change to function that creates this structure to actually set the value from the header


```suggestion
	UserID                 string			`json:"id,omitempty"`
```

Reply:
1. Export means I have to add the UserID to where returning the InputUser.
2. **"you must change to a function that creates this structure to set the value from the header"**- where do i have to do this?

Reply:
> 1. Export means I have to add the UserID to where returning the InputUser. Right?

An exported field in Go is a field that starts [with an upper case letter](https://tip.golang.org/doc/effective_go#names), my suggestion fixes both of the issues related to JSON serialization, I exported the field and changed how it is translated to JSON (simply as `id` since you'll be use it as `input.user.id`)

> 2. "you must change to a function that creates this structure to set the value from the header"- where do i have to do this?

The function is `InputFromRequest` and you can find it in the [`core/input.go`](https://github.com/rond-authz/rond/blob/main/core/input.go#L115C6-L115C22) file.

Here is where you can find the user creation part:

https://github.com/rond-authz/rond/blob/67adb740953c2cd5e86ac4d9ceefb1b70ffd5ee4/core/input.go#L151-L156

Reply:
will it work **UserID: user.id,**?

Reply:
The `user` type is [here](https://github.com/rond-authz/rond/blob/67adb740953c2cd5e86ac4d9ceefb1b70ffd5ee4/types/rbactypes.go#L21) and has a `UserID` field; however I don't remember if it gets properly mapped.

I'd try setting up a test to find out; If it does not get its value you'd have to take the user id from the request headers (you can find the header key from this [environment variable](https://github.com/rond-authz/rond/blob/67adb740953c2cd5e86ac4d9ceefb1b70ffd5ee4/internal/config/env.go#L49C2-L49C14))